src/main/java/com/example/ajaxonboardingproject/config/GrpcServer.kt

```diff
@@ -1,34 +1,14 @@
 package com.example.ajaxonboardingproject.config
 
-import com.example.ajaxonboardingproject.grpc.CinemaHallGrpcService
-import com.example.ajaxonboardingproject.grpc.MovieGrpcService
-import com.example.ajaxonboardingproject.grpc.MovieSessionGrpcService
 import io.grpc.Server
-import io.grpc.ServerBuilder
-import jakarta.annotation.PostConstruct
-import org.springframework.beans.factory.annotation.Value
-import org.springframework.context.annotation.Bean
 import org.springframework.stereotype.Component
 
 @Component
 class GrpcServer(
-    private val cinemaHallGrpcService: CinemaHallGrpcService,
-    private val movieGrpcService: MovieGrpcService,
-    private val movieSessionGrpcService: MovieSessionGrpcService,
-    @Value("\${spring.grpc.port}")
-    var grpcPort: Int
+    private val grpcServer: Server
```

Comment:
you don't need to have this class since you're creating a bean through java config. regarding shutdown, I meant use it in bean config     @Bean(destroyMethod = "shutdown")